utils/cmut__utils__compile_resource.cmake

```diff
@@ -0,0 +1,139 @@
+
+
+include("${CMAKE_CURRENT_LIST_DIR}/../cmut_message.cmake")
+
+
+function(cmut__utils__get_variable_name VARIABLE_NAME INPUT_FILE)
+    # get the filename and replace all ' ' and '.' with '_'
+    get_filename_component(SOURCE_FILE_NAME ${INPUT_FILE} NAME)
+    string(REGEX REPLACE "[^a-zA-Z0-9\-]" "_" SOURCE_FILE_NAME_VARIABLE "${SOURCE_FILE_NAME}")
+    # create Name ouf output files .h & .cpp
+    set(${VARIABLE_NAME} "${SOURCE_FILE_NAME_VARIABLE}" PARENT_SCOPE)
+endfunction()
+
+function(cmut__utils__get_header_name FILE_NAME INPUT_FILE OUTPUT_FOLDER)
+    cmut__utils__get_variable_name(SOURCE_FILE_NAME_VARIABLE "${INPUT_FILE}")
+    # create Name ouf output files .h & .cpp
+    if ("${OUTPUT_FOLDER}" STREQUAL "")
+        set(${FILE_NAME} "GENREATED__${SOURCE_FILE_NAME_VARIABLE}.h" PARENT_SCOPE)
+    else ()
+        set(${FILE_NAME} "${OUTPUT_FOLDER}/GENREATED__${SOURCE_FILE_NAME_VARIABLE}.h" PARENT_SCOPE)
+    endif ()
+endfunction()
+
+function(cmut__utils__get_source_name FILE_NAME INPUT_FILE OUTPUT_FOLDER)
```

Comment:
factorize cmut__utils__get_source_name and cmut__utils__get_header_name

Reply:
done